Source/WebCore/platform/graphics/texmap/coordinated/CoordinatedBackingStoreProxy.cpp

```diff
@@ -148,14 +153,25 @@ OptionSet<CoordinatedBackingStoreProxy::UpdateResult> CoordinatedBackingStorePro
         result.add(UpdateResult::TilesPending);
 
     // Update the dirty tiles.
+    IntRect tileDirtyRectUnion;
     unsigned dirtyTilesCount = 0;
     for (const auto& tile : m_tiles.values()) {
-        if (tile.isDirty())
-            dirtyTilesCount++;
+        if (!tile.isDirty())
+            continue;
+
+        tileDirtyRectUnion.unite(tile.dirtyRect);
+        ++dirtyTilesCount;
     }
 
     WTFBeginSignpost(this, UpdateTiles, "dirty tiles: %u", dirtyTilesCount);
 
+#if USE(SKIA)
+    // Record only once the whole layer.
+    std::optional<Ref<SkiaRecordingResult>> recording;
```

Comment:
I would use RefPtr instead of optional Ref

Reply:
Fair enough.